### ISSUE TITLE
CC3D GY-63, ms5611 (plastic case) baro not working. 1.8.1, 1.9.0 Flexi Port.

### ISSUE THREAD

**Issue by w1ne** [cleanflight/cleanflight#1284]
I tried GY-63 baro board with CC3D without any success, both cleanflight 1.8.1, 1.9.0.
Baro works with arduino.
Also, unsoldering 22K pull up resistors makes no difference.
Logic analyzer i2c sequence included below.
![00](https://cloud.githubusercontent.com/assets/14119286/9675104/b545783a-52b9-11e5-98cc-b9561d2e334c.png)
![0](https://cloud.githubusercontent.com/assets/14119286/9675109/bb1b6468-52b9-11e5-8268-854ca8672723.png)
![1](https://cloud.githubusercontent.com/assets/14119286/9675114/c4ee545a-52b9-11e5-9d72-57d188a1a0d9.png)

**Comment by w1ne** [comment]
i2C with arduino
![3](https://cloud.githubusercontent.com/assets/14119286/9683254/b64ece5c-530e-11e5-8057-de53d47a1ce7.png)
![4](https://cloud.githubusercontent.com/assets/14119286/9683256/b9d6c886-530e-11e5-866b-da4470088a2e.png)
![5](https://cloud.githubusercontent.com/assets/14119286/9683260/bcd15d12-530e-11e5-8f6e-ccb1a2ec1113.png)
Baro works properly.

**Comment by w1ne** [comment]
Also tried to connect baro with magnetometer in parallel. No success.
Magnetometer working but not baro. The board simply not getting any ACK responses from ms5611.
I am using stm32f103c8t6 version of the CC3D board. Any suggestions?

**Comment by SteveAmor** [comment]
You can't add an external baro without the EOC connection as well as I2C which I fixed in #1033 
If #1033 resolves your issue, can you please confirm and close this issue.

**Comment by w1ne** [comment]
@SteveAmor I have no experience with git, how can I pull your fix? I will compile and test.

**Comment by w1ne** [comment]
I tried to compile code from SteveAmor  repository. Problem persists. Anyway, code is related to Acro board and BMP085, so do not know, how it can help.

**Comment by w1ne** [comment]
Ok. I managed get it working by adding dirty hack.
In bus_i2c_stm32f10x.c.
Still do not now the cause. Need to do more research.  But now it works.

    // Generate a start then stop condition
    // SCL  PB10
    // SDA  PB11
    digitalLo(gpio, sda); // Set bus data low
    delayMicroseconds(10);
    digitalLo(gpio, scl); // Set bus scl low
    delayMicroseconds(10);
    digitalHi(gpio, scl); // Set bus scl high
    delayMicroseconds(10);
    digitalHi(gpio, sda); // Set bus sda high
    delayMicroseconds(10);
    digitalLo(gpio, sda); // Set bus data low
    delayMicroseconds(10);
    digitalLo(gpio, scl); // Set bus scl low
    delayMicroseconds(10);
    digitalHi(gpio, scl); // Set bus scl high
    delayMicroseconds(10);
    digitalHi(gpio, sda); // Set bus sda high
    delayMicroseconds(10);
    digitalLo(gpio, sda); // Set bus data low
    delayMicroseconds(10);
    digitalLo(gpio, scl); // Set bus scl low
    delayMicroseconds(10);
    digitalHi(gpio, scl); // Set bus scl high
    delayMicroseconds(10);
    digitalHi(gpio, sda); // Set bus sda high
    delayMicroseconds(10);
